modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/ActivationKeysTable/components/ActionButton/index.js

```diff
@@ -31,8 +32,15 @@ const ActionButton = ({
 	setStatus,
 }) => {
 	const {provisioningServerAPI} = useAppPropertiesContext();
+	const {data} = useGetKoroneikiAccounts();
 	const navigate = useNavigate();
 
+	const dataItems = data.c.koroneikiAccounts.items || [];
+
+	const provisioningResults = dataItems.map(
+		({allowSelfProvisioning}) => allowSelfProvisioning
+	);
```

Comment:
This could be different.

1. Is mapping a value, but it's used to know if provisioningResults exist or not, on [getActivationKeysActionsItems.js](https://github.com/pt-liferay-solutions/liferay-portal/pull/3309/files#diff-af25bfd124ce4e10a1b40c44848bf2cbb627ee90d3e44b89bef0607efdf5d207) line 28;

2. Using `if (provisioningResults)` will always be `TRUE` no matter if has value ir not. this can cause a bug.

3. The better in this case, is transform in a boolean, but it depends on the outcome you expected.

For example, if you need to know, at least ONE item of dataItems.allowSelfProvisioning is true, you can use **SOME**, like this: 

```js
const allowSelfProvisioning = koroneikiAccounts.some(
 		({allowSelfProvisioning}) => allowSelfProvisioning
 );
```

This condition will return a boolean.

If you want to **ALL** items in this array to be true, you can use **EVERY** function.


```js
const allowSelfProvisioning = koroneikiAccounts.every(
 		({allowSelfProvisioning}) => allowSelfProvisioning
 );
```

Map in this case, looks like is wrong placed.

My other suggestions will derive this comment, pay attention.

Reply:
Oh, pick one (some or every) to follow up